libr/core/rvc.c

```diff
@@ -0,0 +1,115 @@
+#include <rvc.h>
+static bool copy_commits(const Rvc *repo, const char *dpath, const char *sname) {
+	char *path, *name, *spath;
+	spath = r_str_newf ("%s" R_SYS_DIR "branches" R_SYS_DIR "%s" R_SYS_DIR "commits", repo->path, sname);
+	r_return_val_if_fail (spath, false);
+	RList *files = r_sys_dir (spath);
+	if (!files) {
+		free (spath);
+		return false;
+	}
+	RListIter *iter;
+	ls_foreach (files, iter, name) {
+		path = r_str_newf ("%s%s", spath, sname);
+		if (!path) {
+			free (spath);
+			r_list_free (files);
+			return false;
+		}
+		if (!r_file_copy (dpath, path)) {
+			free (spath);
+			free (path);
+			r_list_free (files);
+			return false;
+		}
+		free (path);
+	}
+	free (spath);
+	r_list_free (files);
+	return true;
+}
+
+static char *branch_mkdir(Rvc *repo, RvcBranch *b) {
+	char *path = r_str_newf ("%s" R_SYS_DIR "branches" R_SYS_DIR"%s" R_SYS_DIR "commits" R_SYS_DIR,repo->path, b->name);
+	r_return_val_if_fail (path, NULL);
+	if (!r_sys_mkdirp (path)) {
+		free (path);
+		return NULL;
+	}
+	return path;
+}
+
+R_API bool rvc_branch(Rvc *repo, const char *name, const RvcBranch *parent) {
+	char *bpath;
+	RvcBranch *nb = R_NEW0 (RvcBranch);
+	if (!nb) {
+		eprintf ("Failed To Allocate Branch Struct\n");
+		return false;
+	}
+	nb->head = NULL;
+	nb->name = r_str_new (name);
+	if (!nb->name) {
+		eprintf ("Failed To Allocate Branch Name\n");
+		free (nb);
+		return false;
+	}
+	if (!r_list_append (repo->branches, nb)) {
+		eprintf ("Failed To Allocate Branch Struct\n");
+		free (nb->name);
+		free (nb);
+		return false;
+	}
+	bpath = branch_mkdir (repo, nb);
+	if (!bpath) {
+		eprintf ("Failed To Create Branch Directory\n");
+		free (nb->name);
+		free (nb);
+		r_list_pop (repo->branches);
+		return false;
+	}
+	if (parent) {
+		nb->head = parent->head;
+		if (!copy_commits (repo, parent->name, bpath)) {
+			eprintf ("Failed To Copy Commits From Parent\n");
+			free (nb->name);
+			free (nb);
+			free (bpath);
+			return false;
+		}
+	}
+	free (bpath);
+	return true;
+}
+
+R_API Rvc *rvc_new(const char *path) {
+	Rvc *repo;
+	repo = R_NEW (Rvc);
+	if (!repo) {
+		eprintf ("Failed To Allocate Repoistory Path\n");
+	}
+	repo->path = r_str_newf ("%s" R_SYS_DIR ".rvc" R_SYS_DIR, path);
```

Comment:
Check first if .git exists in this directory and run the equivalent shell commands instead

Reply:
should this be done by the caller?

Reply:
Or do you want me to add a complete section dedecated for the git shell commands?
If you want I can change the functions to check for if git is to be used and use the equivalent git commands